igc.cc

```diff
@@ -348,6 +359,45 @@ void IgcFormat::read()
       }
       prev_tod = tod;
 
+      /*
+       * Parse any extension data present. If no extensions are used (unlikely,
+       * but possible in the case of homebrew flight recorders), then skip the
+       * whole thing.
+      */
+      // TODO: A switch would be better than an if ladder...
+      if (fs_metadata.has_data()) {
+        auto* fsdata = new igc_fsdata;
+        if (global_opts.debug_level >= 7) {
+          printf(MYNAME ": Record: %s\n",ibuf_q.toUtf8().constData());
+        }
+        if (global_opts.debug_level >= 6) {
+          printf(MYNAME ": Adding extension data:");
+        }
+        QHash<QString, short>::const_iterator i;
+        for (i = ext_types_hash.constBegin(); i != ext_types_hash.constEnd(); ++i) {
+          QStringView ext_record_type = QStringView(i.key().mid(4,3)).toString();
+
+          igc_ext_type_t def_type = IgcFormat::get_ext_type(ext_record_type.toString());
+          igc_defn_t ext_defn = fs_metadata.get_defn(def_type);
```

Comment:
Can we just use the value of the hash instead?
igc_defn_t ext_defn = fs_metadata.get_defn(i.value());

If so, what is the purpose of the hash if we aren't looking up values from keys?

If we aren't using the hash to get values from keys, and we only really need the values (or the keys) then
Can ext_types_hash be replaced by a QList?